gen3-load-tests/download_performance_test.py

```diff
@@ -0,0 +1,2226 @@
+
+except ImportError as e:
+    GEN3_SDK_AVAILABLE = False
+    print(f"Warning: Gen3 SDK not available: {e}")
+    print("Will skip Gen3 SDK async testing")
+except Exception as e:
+    GEN3_SDK_AVAILABLE = False
+    print(f"Error importing Gen3 SDK: {e}")
+    print("Will skip Gen3 SDK async testing")
+
+
+class PerformanceTimer:
+    def __init__(self):
+        self.timings = {}
+
+    @contextmanager
+    def time_operation(self, operation_name: str):
+        """Context manager to time an operation."""
+        start_time = time.time()
+        print(f"⏱️  Starting: {operation_name}")
+        try:
+            yield self
+        finally:
+            end_time = time.time()
+            duration = end_time - start_time
+            if operation_name not in self.timings:
+                self.timings[operation_name] = []
+            self.timings[operation_name].append(duration)
+            print(f"⏱️  Completed: {operation_name} - {duration:.3f}s")
+
+    def get_timing_summary(self) -> Dict[str, Dict[str, float]]:
+        """Get summary statistics for all timed operations."""
+        summary = {}
+        for operation, times in self.timings.items():
+            if times:
+                summary[operation] = {
+                    "total_time": sum(times),
+                    "avg_time": mean(times),
+                    "min_time": min(times),
+                    "max_time": max(times),
+                    "count": len(times),
+                    "std_dev": stdev(times) if len(times) > 1 else 0.0,
+                }
+        return summary
+
+    def print_summary(self):
+        """Print timing summary to console."""
+        print("\n" + "=" * 60)
+        print("📊 DETAILED PERFORMANCE TIMING BREAKDOWN")
+        print("=" * 60)
+
+        summary = self.get_timing_summary()
+        for operation, stats in summary.items():
+            print(f"\n🔹 {operation}:")
+            print(f"   Total Time: {stats['total_time']:.3f}s")
+            print(f"   Average: {stats['avg_time']:.3f}s")
+            print(f"   Min/Max: {stats['min_time']:.3f}s / {stats['max_time']:.3f}s")
+            if stats["count"] > 1:
+                print(f"   Std Dev: {stats['std_dev']:.3f}s")
+            print(f"   Executions: {stats['count']}")
+
+
+@dataclass
+class TestConfiguration:
+    num_runs: int = 2
+    enable_profiling: bool = True
+    enable_real_time_monitoring: bool = True
+    monitoring_interval: float = 1.0
+    filter_medium_files: bool = False
+    force_uncompressed_cdis: bool = True
+    auto_extract_cdis: bool = True
+    show_detailed_timing: bool = True
+
+    max_concurrent_requests_async: int = 100
+    num_workers_cdis: int = 4
+
+    test_methods: List[str] = field(
+        default_factory=lambda: [
+            "async",
+            "cdis",
+        ]
+    )
+
+    gen3_client_path: str = "gen3-client"
+    gen3sdk_path: str = "~/path/to/gen3sdk-python"  # TODO: Update this path
```

Comment:
we should not rely on this path, refactor to remove this requirement. rely on gen3 being available and imported appropriately